src/sentry/utils/appleconnect/itunes_connect.py

```diff
@@ -302,9 +302,26 @@ def _request_trusted_phone_info(self) -> None:
             },
             timeout=REQUEST_TIMEOUT,
         )
-        if response.status_code != HTTPStatus.OK:
-            raise ITunesError(f"Unexpected response status: {response.status_code}")
-        info = response.json()["trustedPhoneNumber"]
+        # Fastlane actually doesn't check the response status in their implementation and simply
+        # checks whether the response returns the expected fields associated with 2FA. This is
+        # tries to be lenient in a similar way.
+        is_expected_status = (
+            response.status_code == HTTPStatus.OK or response.status_code == HTTPStatus.CREATED
```

Comment:
Can we just check for `response.ok`?

Reply:
i've opted to just check for `response.ok` and short-circuit if that's false - i figure there's nothing wrong with being a little bit more cautious there in case an error-like response does return `trustedPhoneNumber`.